### ISSUE TITLE
[Question] Method to pass environment flag to GAE computation

### ISSUE THREAD

**Issue by SeanDaSheep** [DLR-RM/stable-baselines3#1390]
### ❓ Question

Can the returns of Generalized Advantage Estimation be changed during steps with env variables? (e.g. reset to 0 if a flag is True in the environment). What I mean here is if I can pass values to the rollout buffer. In my use case, I'm training using PPO and using a custom env. I apologize if my understanding of this topic is not sufficient.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by SeanDaSheep** [comment]
tldr, what I'm asking here is if I can create a custom rollout buffer similar to a custom callback

**Comment by araffin** [comment]
Hello,
yes but you need to fork SB3 for that.
You may also have a look at https://github.com/HumanCompatibleAI/imitation

**Comment by SeanDaSheep** [comment]
Modifying the function in RolloutBuffer from buffer.py has worked as expected. Thanks for the help.
